batch7crud-roll4/src/main/java/com/lftechnology/batch7crud/dao/EmployeeDao.java

```diff
@@ -0,0 +1,232 @@
+
+      rs = stmt.executeQuery();
+
+      while (rs.next()) {
+        Employee e = new Employee();
+        e.setId(rs.getInt("id"));
+        e.setFirstName(rs.getString("first_name"));
+        e.setLastName(rs.getString("last_name"));
+        e.setStation(rs.getString("station"));
+        employeeList.add(e);
+      }
+      return employeeList;
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+
+    } finally {
+      try {
+        if (rs != null)
+          rs.close();
+        if (stmt != null)
+          stmt.close();
+        if (connection != null)
+          connection.close();
+
+      } catch (SQLException e) {
+        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      }
+    }
+  }
+
+  public Employee fetchById(int id) throws DataException {
+    Connection connection = null;
+    PreparedStatement stmt = null;
+    ResultSet rs = null;
+
+    try {
+      connection = DbConnection.getConnection();
+      Employee e = null;
+      String sql = "SELECT * FROM employees WHERE id=?";
+      stmt = connection.prepareStatement(sql);
+      stmt.setInt(1, id);
+      rs = stmt.executeQuery();
+
+      while (rs.next()) {
+        e = new Employee();
+        e.setId(rs.getInt("id"));
+        e.setFirstName(rs.getString("first_name"));
+        e.setLastName(rs.getString("last_name"));
+        e.setStation(rs.getString("station"));
+      }
+      return e;
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+
+    } finally {
+      try {
+        if (rs != null)
+          rs.close();
+        if (stmt != null)
+          stmt.close();
+        if (connection != null)
+          connection.close();
+
+      } catch (SQLException e) {
+        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      }
+    }
+  }
+
+  public void insert(Employee employee) throws DataException {
+    Connection connection = null;
+    PreparedStatement stmt = null;
+
+    try {
+      connection = DbConnection.getConnection();
+      String sql = "INSERT INTO employees(first_name, last_name, station) VALUES(?,?,?)";
+      stmt = connection.prepareStatement(sql);
+      stmt.setString(1, employee.getFirstName());
+      stmt.setString(2, employee.getLastName());
+      stmt.setString(3, employee.getStation());
+
+      stmt.executeUpdate();
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+
+    } finally {
+      try {
+        if (stmt != null)
+          stmt.close();
+        if (connection != null)
+          connection.close();
+      } catch (SQLException e) {
+        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      }
+    }
+  }
+
+  public void update(Employee employee) throws DataException {
+    Connection connection = null;
+    PreparedStatement stmt = null;
+    ResultSet rs = null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "rs" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)